FlashCards.ZubePierreBasali/StudySessions/StudyController.cs

```diff
@@ -0,0 +1,241 @@
+                int? totalQuestions = 0;
+                int totalScores = 0;
+                TimeSpan totalTime = new();
+
+                Table finalTable = new();
+                finalTable.Expand();
+                finalTable.Centered();
+                finalTable.AddColumn(stack);
+                finalTable.Columns[0].Centered();
+                finalTable.Border(TableBorder.Rounded);
+
+                Table table = new();
+                table.Centered();
+                table.Border(TableBorder.Rounded);
+                table.Title("All Sessions");
+                table.AddColumns("Date","Question Mode", "Number Of Questions", "Score", "Time");
+
+                foreach (StudyModel session in sessions)
+                {
+                    if (session.Time == "00:00:00") session.Time = "Off";
+                    table.AddRow(session.Date,session.QuestionMode, session.QuestionCount.ToString(), session.Score.ToString(), session.Time);
+                    TimeSpan timeToAdd;
+                    bool addTime = TimeSpan.TryParseExact(session.Time, "c", CultureInfo.InvariantCulture, TimeSpanStyles.None,out timeToAdd);
+                    if ((addTime))
+                    {
+                    totalTime = totalTime.Add(TimeSpan.ParseExact(session.Time, "c", CultureInfo.InvariantCulture, TimeSpanStyles.None));
+                    }
+                    totalQuestions += session.QuestionCount;
+                    totalScores += session.Score;
+                }
+                finalTable.AddRow(table);
+
+                table = new();
+                table.Centered();
+                table.Border(TableBorder.AsciiDoubleHead);
+                table.Title("Resume");
+                table.AddColumns("Total Number of Sessions", "Total Number Of Questions", "Total Score", "Total Time");
+                table.AddRow(sessions.Count.ToString(), totalQuestions.ToString(), totalScores.ToString(), totalTime.ToString());
+                finalTable.AddRow(table);
+
+                table = new();
+                table.Centered();
+                table.Border(TableBorder.DoubleEdge);
+                table.Title($"Monthly Resume");
+                table.AddColumn("Year-Month");
+
+                List<StudyMonthly> monthly = dataTools.GetMonthlyReports(stack);
+                int maxIndex = monthly.Count;
+                for(int i = 0; i < maxIndex; i++) table.AddColumn(monthly[i].Month);
+
+                table.AddRow("Number of Session");    
+                for (int i = 0; i < maxIndex; i++) table.Rows.Update(0, i+1, new Markup($"{monthly[i].QuestionCount}"));
+
+                finalTable.AddRow(table);
+                AnsiConsole.Write(finalTable);
+                AnsiConsole.WriteLine();
+            }
+        }
+
+        internal  void CreateStudySessionsData(DataTools dataTools)
+        {
+            Random random = new();
```

Comment:
🟡 **Random.Shared**

📓 You can use `Random.Shared` so you don't need to initialise a new `Random` object.